nntrainer/graph/network_graph.cpp

```diff
@@ -181,13 +190,19 @@ int NetworkGraph::realizeMultiInputType(Layer &current) {
   if (current.getNumInputs() == 1)
     return ML_ERROR_NONE;
 
+  // TODO: this can be addition or concat layer - add support
   std::shared_ptr<Layer> layer = nntrainer::createLayer(AdditionLayer::type);
   ensureName(layer, current.getName());
+
   layer->setNumInputs(current.getNumInputs());
   layer->input_layers.clear();
   for (unsigned int i = 0; i < current.input_layers.size(); ++i)
     layer->input_layers.push_back(current.input_layers[i]);
 
+  layer->setNumOutputs(1);
+  layer->output_layers.clear();
+  layer->output_layers.push_back(current.getName());
```

Comment:
setNumOutputs is called when the layer is created. You can remove this. 
Is it dummy? set the layer->output_layers with current.getName()?? it is not actually. it could be the output of the current layer.

Reply:
In this case, `multiInputLayer` is added as utility to the current layer so that it can merge (add or concat) all its inputs and provide it as a single input to the `current` layer. So, the `output_layer` of the new `layer` must be `current`. Please let me know if I got the understanding wrong.

`setNumOutputs(1)` is done for clarity and reading purposes (you are right that it is initialized to 1 anyway, so I think compiler will optimize it out as it is a compile time constant).

Reply:
right. BTW, Isn't it set at NetworkGraph::setOutputLayers()? it is ok to set twice??

Reply:
`setOutputLayers()` is added here just for readability. Yes, it is alright to set twice or reset it.